parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java

```diff
@@ -503,8 +503,7 @@ protected Builder(OutputFile path) {
      * @return an appropriate WriteSupport for the object model.
      */
     protected WriteSupport<T> getWriteSupport(ParquetConfiguration conf) {
-      throw new UnsupportedOperationException(
```

Comment:
When people want to decouple from Hadoop, they can just override the methods, otherwise the old behavior is preserved.

Reply:
Thinking of it, I believe we should deprecate the ones where you need to pass in Configuration (from Hadoop), and then we can move to `ParquetConfiguration` where you can also pass in a `HadoopParquetConfiguration`. Thanks for raising this, let me create a commit and let me know what you think!